### PR TITLE
Remove unnecessary getattr/hasattr usage

### PR DESCRIPTION
**Pull request by zas**
## Summary

* This is a…
  * [ ] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

Replace unnecessary `getattr()`/`hasattr()` calls with direct attribute access where the attribute is guaranteed to be initialized.

## Problem

Several places in the codebase use defensive `getattr(obj, attr, default)` or `hasattr(obj, attr)` patterns on attributes that are always initialized in `__init__`. This hides potential bugs (typos in attribute names silently return defaults instead of raising `AttributeError`), reduces readability, and in one case contained dead fallback code referencing non-existent NamedTuple fields.

## Solution

Each commit addresses one file/component:
- **tagger.py**: `_restoring_session` (always set in `__init__`)
- **options/player**: `window.player` (always initialized on MainWindow)
- **itemviews**: `file.acoustid_fingerprint` and `Cluster.special` (always initialized)
- **plugin3/installable**: `self.description` (always set in `__init__`)
- **plugin3/manager**: `source.resolved_ref_type`, `PluginMetadata.url`, `PluginManifest._data`
- **plugindetailswidget**: `PluginManifest` methods (always exist)
- **pluginlistwidget**: `UpdateResult` fields (NamedTuple, always present) + dead fallback code removed

## AI Usage

In accordance with the AI use policy portion of the [MetaBrainz Contribution Guidelines](https://github.com/metabrainz/guidelines/blob/master/README.md), the level of AI/LLM use in the development of this Pull Request is:

* [ ] No AI/LLM use
* [ ] Minimal use (e.g. autocompletion)
* [ ] Moderate use (e.g. suggestions regarding code fragments)
* [x] Significant use (e.g. code structure, tests development, etc.)
* [ ] Primarily AI developed
* [ ] Other (please specify below)

AI identified candidates and proposed changes; human reviewed each for correctness (init order, type guarantees, edge cases).

## Action

Additional actions required:
* [ ] Update Picard [documentation](https://github.com/metabrainz/picard-docs) (please include a reference to this PR)
* [ ] Other (please specify below)

### CODE REVIEW

File: picard/ui/mainwindow/__init__.py [hunk 1]
```diff
@@ -348,7 +348,7 @@ def _setup_player(self):
 
     def update_now_playing_services(self):
         config = get_config()
-        for service, config_key in getattr(self, '_now_playing_services', []):
```
Comment (phw) [inline picard/ui/mainwindow/__init__.py:351]: The `_now_playing_services` is actually conditionally set. If `disable_player` is True in `MainWindow` it is not even initialized. We could of course always initialize it.

**Review by phw — CHANGES_REQUESTED** [review]

File: picard/ui/mainwindow/__init__.py (hunk 1 above)
Comment (zas) [reply picard/ui/mainwindow/__init__.py:351]: You're right, dropped the commit and rebased.

**Review by phw — APPROVED** [review]
